pkg/injection/codemodule/installer/image/installer.go

```diff
@@ -30,15 +30,14 @@ type Properties struct {
 }
 
 func NewImageInstaller(ctx context.Context, fs afero.Fs, props *Properties) (installer.Installer, error) {
-	pullSecret := props.Dynakube.PullSecretWithoutData()
 	defaultTransport := http.DefaultTransport.(*http.Transport).Clone()
 
 	transport, err := registry.PrepareTransportForDynaKube(ctx, props.ApiReader, defaultTransport, props.Dynakube)
 	if err != nil {
 		return nil, err
 	}
 
-	keychain, err := dockerkeychain.NewDockerKeychain(ctx, props.ApiReader, pullSecret)
+	keychain, err := dockerkeychain.NewDockerKeychains(ctx, props.ApiReader, props.Dynakube.Namespace, props.Dynakube.PullSecretsNames())
```

Comment:
you could pass the dynakube here instead of 2 parameters, since you can access PullSecrets and Namespace inside the function

Reply:
you shouldn't, that will make it bound to the DynaKube, what if you want to use it for EdgeConnect?
or in a case where you don't have the whole DynaKube?

passing around the DynaKube is nice and easy, but lets not make it become a kind of a "singleton" that we pass around in every helper package and function

Reply:
you could do some renaming, there could be a `dockerkeychain.New` (so we don't stutter) where you request just what you need, and then introduce a `dockerkeychain.NewForDynaKube()`, where you would do what @luhi-DT was requesting. (but IMO `dockerkeychain.NewForDynaKube()` feels a bit of an overkill, but up to you)